apps/rush-lib/src/cli/actions/CustomRushAction.ts

```diff
@@ -0,0 +1,175 @@
+      throw new Error(`File not found: ${this.rushConfiguration.rushLinkJsonFilename}` +
+        `${os.EOL}Did you run "rush link"?`);
+    }
+    this.eventHooksManager.handle(Event.preRushBuild);
+
+    const stopwatch: Stopwatch = Stopwatch.start();
+
+    const isQuietMode: boolean = !(this._verboseParameter.value);
+
+    // collect all custom flags here
+    const customFlags: string[] = [];
+    this.customOptions.forEach((customOption: ICustomOptionInstance) => {
+      if (customOption.parameterValue.value) {
+        if (customOption.optionType === 'flag') {
+          customFlags.push(customOption.longName);
+        } else if (customOption.optionType === 'enum') {
+          customFlags.push(`${customOption.longName}=${customOption.parameterValue.value}`);
+        }
+      }
+    });
+
+    const tasks: TaskManager = new TaskManager(
+      this._parser.rushConfig,
+      this._toFlag.value,
+      this._fromFlag.value,
+      this.options.actionVerb,
+      customFlags,
+      isQuietMode,
+      this._parallelismParameter.value,
+      this.options.actionVerb === 'build'
+    );
+
+    tasks.execute().then(
+      () => {
+        stopwatch.stop();
+        console.log(colors.green(`rush ${this.options.actionVerb} (${stopwatch.toString()})`));
+        this._collectTelemetry(stopwatch, true);
+        this._parser.flushTelemetry();
+        this.eventHooksManager.handle(Event.postRushBuild, this._parser.isDebug);
+      },
+      () => {
+        stopwatch.stop();
+        console.log(colors.red(`rush ${this.options.actionVerb} - Errors! (${stopwatch.toString()})`));
+        this._collectTelemetry(stopwatch, false);
+        this._parser.flushTelemetry();
+        this.eventHooksManager.handle(Event.postRushBuild, this._parser.isDebug);
+        this._parser.exitWithError();
+      });
+  }
+
+  protected onDefineParameters(): void {
+    this._parallelismParameter = this.defineIntegerParameter({
+      parameterLongName: '--parallelism',
+      parameterShortName: '-p',
+      key: 'COUNT',
+      description: 'Change limit the number of simultaneous builds. This value defaults to the number of CPU cores'
+    });
+    this._toFlag = this.defineStringListParameter({
+      parameterLongName: '--to',
+      parameterShortName: '-t',
+      key: 'PROJECT1',
+      description: 'Build the specified project and all of its dependencies'
+    });
+    this._fromFlag = this.defineStringListParameter({
+      parameterLongName: '--from',
+      parameterShortName: '-f',
+      key: 'PROJECT2',
+      description: 'Build all projects that directly or indirectly depend on the specified project'
+    });
+    this._verboseParameter = this.defineFlagParameter({
+      parameterLongName: '--verbose',
+      parameterShortName: '-v',
+      description: 'Display the logs during the build, rather than just displaying the build status summary'
+    });
+
+    // @TODO we should throw if they are trying to overwrite built in flags
```

Comment:
>// @TODO  [](start = 4, length = 9)

?

Reply:
we just let ts-command-line throw if they do this.

---
In reply to: [146732263](https://github.com/Microsoft/web-build-tools/pull/387#discussion_r146732263) [](ancestors = 146732263)